brickBreaker.js

```diff
@@ -0,0 +1,166 @@
+class BrickBreaker {
+  constructor() {
+    this.canvas = document.getElementById("canvas");
+    this.ctx = this.canvas.getContext("2d");
+    this.score = 0;
+    this.leftArrowPressed = false;
+    this.rightArrowPressed = false;
+    this.initializeEntities();
+    this.initializeEventListeners();
+  }
+
+  initializeEntities() {
+    this.ball = new Ball(this.ctx, this.canvas.width / 2, 200, 10, "#FFF", 4);
+    this.paddle = new Paddle(this.ctx, 100, 10, 150, 525, 5, "grey");
+    this.bricks = [];
+
+    const colors = ["red", "blue", "green", "yellow", "purple"];
+
+    let y = 0;
+    for (let row = 0; row < 5; row++) {
+      let x = 0;
+      if (!this.bricks[row]) {
+        this.bricks[row] = [];
+      }
+      for (let col = 0; col < 8; col++) {
+        let brick = new Brick(this.ctx, x, y, 50, 25, colors[row]);
```

Comment:
instead of extra variables x and y, why not just use row and col?

`let brick = new Brick(this.ctx, row * 50, col * 25, 50, 25, colors[row]);`

Reply:
row * 50 and col * 25 would not work as x and y since the x value is reset after every row so that the first brick starts from the left edge of the canvas.

Reply:
the reason why you reset x to 0 is because your for loop is iterating row by column (outer loop is for rows, inner loop for col). this means that if you were to populate the grid, you'd be traversing it left to right, and then you need to reset it back to 0 so restart for the next row.

instead, you should be iterating column by row (outer loop column, inner loop row) so that it traverses the grid top to bottom, without needing to manually reset any variable itself.